includes/Gateway/Cash_App_Pay_Gateway.php

```diff
@@ -0,0 +1,1226 @@
+
+				// store the guests customers in our index to avoid future duplicates
+				if ( ! $order->get_user_id() ) {
+					Customer_Helper::add_customer( $order->square_customer_id, $order->get_billing_email() );
+				}
+			} catch ( \Exception $exception ) {
+
+				// log the error, but continue with payment
+				if ( $this->debug_log() ) {
+					$this->get_plugin()->log( $exception->getMessage(), $this->get_id() );
+				}
+			}
+		}
+
+		return $order;
+	}
+
+	/**
+	 * Gets an order with refund data attached.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param int|\WC_Order $order order object
+	 * @param float $amount amount to refund
+	 * @param string $reason response for the refund
+	 *
+	 * @return \WC_Order|\WP_Error
+	 */
+	protected function get_order_for_refund( $order, $amount, $reason ) {
+
+		$order                 = parent::get_order_for_refund( $order, $amount, $reason );
+		$order->square_version = $this->get_order_meta( $order, 'square_version' );
+		$transaction_date      = $this->get_order_meta( $order, 'trans_date' );
+
+		if ( $transaction_date ) {
+			// refunds with the Refunds API can be made up to 1 year after payment and up to 120 days with the Transactions API
+			$max_refund_time = version_compare( $order->square_version, '2.2', '>=' ) ? '+1 year' : '+120 days';
+
+			// throw an error if the payment cannot be refunded
+			if ( time() >= strtotime( $max_refund_time, strtotime( $transaction_date ) ) ) {
+				/* translators: %s maximum refund date. */
+				return new \WP_Error( 'wc_square_refund_age_exceeded', sprintf( __( 'Refunds must be made within %s of the original payment date.', 'woocommerce-square' ), '+1 year' === $max_refund_time ? 'a year' : '120 days' ) );
+			}
+		}
+
+		$order->refund->location_id = $this->get_order_meta( $order, 'square_location_id' );
+		$order->refund->tender_id   = $this->get_order_meta( $order, 'authorization_code' );
+
+		if ( ! $order->refund->tender_id ) {
+
+			try {
+				$response = version_compare( $order->square_version, '2.2', '>=' ) ? $this->get_api()->get_payment( $order->refund->trans_id ) : $this->get_api()->get_transaction( $order->refund->trans_id, $order->refund->location_id );
+
+				if ( ! $response->get_authorization_code() ) {
+					throw new \Exception( 'Tender missing' );
+				}
+
+				$this->update_order_meta( $order, 'authorization_code', $response->get_authorization_code() );
+				$this->update_order_meta( $order, 'square_location_id', $response->get_location_id() );
+
+				$order->refund->location_id = $response->get_location_id();
+				$order->refund->tender_id   = $response->get_authorization_code();
+
+			} catch ( \Exception $exception ) {
+
+				return new \WP_Error( 'wc_square_refund_tender_missing', __( 'Could not find original transaction tender. Please refund this transaction from your Square dashboard.', 'woocommerce-square' ) );
```

Comment:
Is that message something already used by Square or Cash App?  "tender" isn't a term I'd expect to be super familiar and might recommend something else unless this message was provided by Square/Cash App?

Reply:
We can update the message on our side. However, the term '[tender](https://developer.squareup.com/reference/square/objects/Tender) ' is from Square's side, indicating the 'Method of payment.' The Square credit card gateway has this error message, so I followed the same message here." please let me know if you want to this to "method of payment" for both gateways. 

cc: @Sidsector9

Reply:
If that's what's already leveraged with Square, then let's keep `tender`... thanks!